lib/ex_aws/sns/request.ex

```diff
@@ -0,0 +1,23 @@
+defmodule ExAws.SNS.Request do
+  @moduledoc false
+  # SNS specific request logic.
+
+  @empty_body_hash AWSAuth.Utils.hash_sha256("")
+
+  def request(client, action, params) do
+    query = params
+    |> Map.put("Action", Mix.Utils.camelize(Atom.to_string(action)))
+    |> URI.encode_query
```

Comment:
So the issue at the moment is that the AWSAuth library we use for signing at the moment ends up doubly encoding the parameters, because it calls URI.encode as part of its signing process. This is rather annoying, because it means you can't pass it a pre-encoded URL.

I'm pretty sure we want to just move the signing stuff in house at this point, AwsAuth provides a decent example, but suffers from having everything built into a couple massive functions.